### PR TITLE
x86 encoding information

### PR DESCRIPTION
**Pull request by obs1dium**
This PR is based on #520's code with the following changes and additions:
- cs_x86_encode renamed to cs_x86_encoding
- its redundant disp_valid and imm_valid members have been removed
- moved modrm, sib, disp to cs_x86_encoding
- x86_imm_size has been extended to include the encoded immediate size, immediate sizes for bit test and shift/rotate operations have been corrected
- disp has been promoted to 64-bits to accomodate moffset64 operands
- moffset operands will show up as disp in cs_x86_encoding instead of imm

This should address PRs #520, #444, #331 and issues #525 (partially), #521, #509, #443 (partially)

### CODE REVIEW

**Comment by bughoho** [comment]
merged, Great!

File: include/capstone/x86.h [hunk 1]
```diff
@@ -300,23 +308,6 @@ typedef struct cs_x86 {
 	// Address size, which can be overrided with above prefix[5].
 	uint8_t addr_size;
 
-	// ModR/M byte
-	uint8_t modrm;
```
Comment (aquynh) [inline include/capstone/x86.h:304]: this patch moves modrm, sib, disp, sib_index, sib_scale, sib_base, etc into new cs_x86_encoding struct, thus breaks compatibility, and breaks it hard. how about keeping these fields in the old place, and cs_x86_encoding only contains new data xxx_offset, so old code still works?

**Comment by bughoho** [comment]
@obs1dium

File: include/capstone/x86.h (hunk 1 above)
Comment (obs1dium) [inline include/capstone/x86.h:304]: I thought it made more sense to have all the related info in one place. Easy to undo if you want to stay backwards compatible, though.

Comment (aquynh) [inline include/capstone/x86.h:304]: yes, backward compatible is important, as we do not want to force so many apps to be rewritten.

another reason is that i think the encoding struct is solely for encoding info (all those offsets are to help you to find out where to patch the original input), so it makes more sense to me to keep sib/disp/etc in the original place.

thanks.

Comment (obs1dium) [inline include/capstone/x86.h:304]: moved them back

**Pull request merged**